### PR TITLE
fix(Scripts/Items): Fix Goblin Bomb Dispenser

### PR DESCRIPTION
**Pull request by ZhengPeiRu21**
## Changes Proposed:
-  Script was existing but missing malfunction explosion chance and was not assigned correctly
- Some NPC values were incorrect; these are now fixed
- All values are from TrinityCore: https://github.com/TrinityCore/TrinityCore/pull/27749

Co-Authored-By: offl <11556157+offl@users.noreply.github.com>

## Issues Addressed:

- Closes unreported issue

## SOURCE:

TrinityCore: https://github.com/TrinityCore/TrinityCore/pull/27749

## Tests Performed:

Summon appears:
Capture

Attacks nearest target, explodes then dies:
Capture2

Malfunction Explosion working:
Capture3

## How to Test the Changes:

1. .additem 10587
2. Test if it is working

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by acidmanifesto — APPROVED** [review]

File: src/server/scripts/Spells/spell_item.cpp [hunk 1]
```diff
@@ -3384,15 +3385,13 @@ class spell_item_goblin_bomb : public SpellScript
...
+        return ValidateSpellInfo({ SPELL_SUMMON_GOBLIN_BOMB, SPELL_MALFUNCTION_EXPLOSION });
     }
 
     void HandleDummy(SpellEffIndex /*effIndex*/)
     {
         if (Unit* caster = GetCaster())
-        {
-            caster->CastSpell(caster, SPELL_SUMMON_GOBLIN_BOMB, true, GetCastItem());
-        }
+            caster->CastSpell(caster, roll_chance_i(95) ? SPELL_SUMMON_GOBLIN_BOMB : SPELL_MALFUNCTION_EXPLOSION, true, GetCastItem());
```
Comment (IntelligentQuantum) [inline src/server/scripts/Spells/spell_item.cpp:3394]: Bracket

**Review by IntelligentQuantum — CHANGES_REQUESTED** [review]

**Comment by Lynethris** [comment]
Need a test or is it in WIP ?

**Comment by IntelligentQuantum** [comment]
@Lynethris Test please

**Comment by ZhengPeiRu21** [comment]
> Need a test or is it in WIP ?

Sorry for the slow responding. Yes, please test - this is complete (unless any bug are found ^_^).

**Comment by porkserver** [comment]
Tested working
Bomb summons and attacks target and dies/despawns - screenshot shows cooldown active and combat log
Screen Shot 2022-04-04 at 8 48 37 PM
.

**Comment by acidmanifesto** [comment]
@IntelligentQuantum Are the current changes to your liking so we can merge?

**Review by IntelligentQuantum — APPROVED** [review]

**Comment by acidmanifesto** [comment]
Thank you for the PR

**Pull request merged**